File: src/examples/java/org/nodex/examples/net/EchoServer.java
/*
 * Copyright 2002-2011 the original author or authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
 * this file except in compliance with the License. You may obtain a copy of the
 * License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed
 * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations under the License.
 */

package org.nodex.examples.net;

import org.nodex.core.buffer.Buffer;
import org.nodex.core.buffer.DataHandler;
import org.nodex.core.net.NetConnectHandler;
import org.nodex.core.net.NetServer;
import org.nodex.core.net.NetSocket;

public class EchoServer {
  public static void main(String[] args) throws Exception {
    NetServer server = NetServer.createServer(new NetConnectHandler() {
      public void onConnect(final NetSocket socket) {
        System.out.println("Sever conttect");
        socket.data(new DataHandler() {
          public void onData(Buffer buffer) {
            socket.write(buffer);
          }
        });
      }
    }).listen(8080);

    System.out.println("Any key to exit");
    System.in.read();

    server.close();
  }
}

File: src/tests/javascript/net/echo_server.js
//load('core/net.js')
load('test_utils.js')

var tu = new TestUtils();

var server = new vertx.NetServer();

server.connectHandler(function(sock) {
  tu.checkContext();
  sock.dataHandler(function(data) {
    tu.checkContext();
    sock.write(data);
  })
})

server.listen(8080, 'localhost');

tu.appReady();

function vertxStop() {
  tu.checkContext();
  server.close(function() {
    tu.checkContext();
    tu.appStopped();
  });
}
